### DISCUSSION TITLE
How to combine a custom edge with 'ste

### DISCUSSION THREAD

**Discussion by ZakKa89** [xyflow/xyflow#3455] [Q&A] [answered]
https://reactflow.dev/docs/examples/edges/custom-edge/

Once you use a custom edge, which I need because I want to add a button in the middle of my edge, you can't specifiy an edge type anymore.

I would like use the 'step' edge type but with a custom edge. As you can see here the custom edge type is using the default bezier style: 
 
<img width="502" alt="image" src="https://github.com/wbkd/react-flow/assets/24491661/7ad34af1-e1f0-4f07-95f4-81c20a8780e8">

It would be great if the `<BaseEdge />` component supports the edge type as a prop.

**Comment by bcakmakoglu** [comment, marked as answer]
`BaseEdge` takes a `path` prop, which you can generate by using [one of these helpers](https://reactflow.dev/docs/api/edges/edge-utils).

So if you want a smoothstep edge, simply use `getSmoothStepPath`.

**Reply by bcakmakoglu** [reply]
Here's some example code for it:

```tsx
import React from "react";
import {
  BaseEdge,
  EdgeLabelRenderer,
  EdgeProps,
  getSmoothStepPath
} from "reactflow";

import "./buttonedge.css";

const onEdgeClick = (evt, id) => {
  evt.stopPropagation();
  alert(`remove ${id}`);
};

export default function CustomEdge({
  id,
  sourceX,
  sourceY,
  targetX,
  targetY,
  sourcePosition,
  targetPosition,
  style = {},
  markerEnd
}: EdgeProps) {
  const [edgePath, labelX, labelY] = getSmoothStepPath({
    sourceX,
    sourceY,
    sourcePosition,
    targetX,
    targetY,
    targetPosition
  });

  return (
    <>
      <BaseEdge path={edgePath} markerEnd={markerEnd} style={style} />
      <EdgeLabelRenderer>
        <div
          style={{
            position: "absolute",
            transform: `translate(-50%, -50%) translate(${labelX}px,${labelY}px)`,
            fontSize: 12,
            // everything inside EdgeLabelRenderer has no pointer events by default
            // if you have an interactive element, set pointer-events: all
            pointerEvents: "all"
          }}
          className="nodrag nopan"
        >
          <button
            className="edgebutton"
            onClick={(event) => onEdgeClick(event, id)}
          >
            ×
          </button>
        </div>
      </EdgeLabelRenderer>
    </>
  );
}
```

**Reply by ZakKa89** [reply]
Thank you very much! I was so focused finding a prop within BaseEdge that I didn't even realise there was a hook.
